test/integration/evictions/evictions_test.go

```diff
@@ -334,6 +340,85 @@ func TestEvictionVersions(t *testing.T) {
 	}
 }
 
+// TestEvictionWithFinalizers tests eviction with the use of finalizers
+func TestEvictionWithFinalizers(t *testing.T) {
+	cases := map[string]struct {
+		enablePodDisruptionConditions bool
+		phase                         v1.PodPhase
+	}{
+		"terminal pod with PodDisruptionConditions enabled": {
+			enablePodDisruptionConditions: true,
+			phase:                         v1.PodSucceeded,
+		},
+		"terminal pod with PodDisruptionConditions disabled": {
+			enablePodDisruptionConditions: false,
+			phase:                         v1.PodSucceeded,
+		},
+		"running pod with PodDisruptionConditions enabled": {
+			enablePodDisruptionConditions: true,
+			phase:                         v1.PodRunning,
+		},
+		"running pod with PodDisruptionConditions disabled": {
+			enablePodDisruptionConditions: false,
+			phase:                         v1.PodRunning,
+		},
+	}
+	for name, tc := range cases {
+		t.Run(name, func(t *testing.T) {
+			closeFn, rm, informers, _, clientSet := rmSetup(t)
+			defer closeFn()
+
+			ns := framework.CreateNamespaceOrDie(clientSet, "eviction-with-finalizers", t)
+			defer framework.DeleteNamespaceOrDie(clientSet, ns, t)
+			defer featuregatetesting.SetFeatureGateDuringTest(t, feature.DefaultFeatureGate, features.PodDisruptionConditions, tc.enablePodDisruptionConditions)()
+
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+			informers.Start(ctx.Done())
+			go rm.Run(ctx)
+
+			pod := newPod("pod")
+			pod.ObjectMeta.Finalizers = []string{"test.k8s.io/finalizer"}
+			if _, err := clientSet.CoreV1().Pods(ns.Name).Create(ctx, pod, metav1.CreateOptions{}); err != nil {
+				t.Errorf("Failed to create pod: %v", err)
+			}
+
+			pod.Status.Phase = tc.phase
+			addPodConditionReady(pod)
+			if _, err := clientSet.CoreV1().Pods(ns.Name).UpdateStatus(ctx, pod, metav1.UpdateOptions{}); err != nil {
+				t.Fatal(err)
+			}
+
+			waitToObservePods(t, informers.Core().V1().Pods().Informer(), 1, tc.phase)
+			deleteOption := metav1.DeleteOptions{}
+
+			eviction := newV1Eviction(ns.Name, pod.Name, deleteOption)
+
+			err := wait.PollImmediate(5*time.Second, 60*time.Second, func() (bool, error) {
+				e := clientSet.PolicyV1().Evictions(ns.Name).Evict(ctx, eviction)
+				if e != nil {
+					return false, e
+				}
+				return true, nil
+			})
+			if err != nil {
+				t.Fatalf("Eviction of pod failed %v", err)
+			}
+
+			updatedPod, e := clientSet.CoreV1().Pods(ns.Name).Get(ctx, pod.Name, metav1.GetOptions{})
+			if e != nil {
+				t.Fatalf("Failed to get the pod %q with error: %q", klog.KObj(pod), e)
+			}
+			_, cond := podutil.GetPodCondition(&updatedPod.Status, v1.PodConditionType(v1.AlphaNoCompatGuaranteeDisruptionTarget))
+			if tc.enablePodDisruptionConditions == true && cond == nil {
```

Comment:
is there any possibilitiy the test can have a false negative with other condition or not the condition that we are exactly expecting?

Reply:
I think it is reasonable to restrict the checks to the conditions which are involved in the recent code changes (DistruptionTarget in this case). Actually, most of the tests within evictions_test.go add another `Ready` condition, which I think we can skip in the assertions.